app/src/main/java/com/celzero/bravedns/data/AppMode.kt

```diff
@@ -112,17 +142,18 @@ class AppMode internal constructor(private val context: Context,
         return persistentState.braveMode == APP_MODE_DNS
     }
 
-    fun isDnsProxyEnabled(): Boolean {
-        return persistentState.dnsType == PREF_DNS_MODE_PROXY
+    fun isDnsFirewallMode(): Boolean {
+        return persistentState.braveMode == APP_MODE_DNS_FIREWALL
     }
 
     private fun fetchProxyModeFromDb(): Long {
         val dnsProxy = dnsProxyEndpointRepository.getConnectedProxy()
+
         return when (dnsProxy.proxyType) {
-            proxyTypeInternal -> {
+            DNSProxyEndpointRepository.INTERNAL -> {
```

Comment:
If we are creating enums like above, then this could be another enum (may be within this class itself):

```
enum class DnsProxyType {
   INTERNAL, EXTERNAL
}
```

Reply:
these are unused now, removed.